packages/xrpl/src/client/index.ts

```diff
@@ -666,8 +667,41 @@ class Client extends EventEmitter<EventTypes> {
       promises.push(checkAccountDeleteBlockers(this, tx))
     }
 
-    return Promise.all(promises).then(() => tx)
+    // further manipulation of tx_ uses non-SubmittableTransaction types, hence we need a typecast to any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/consistent-type-assertions -- `any` type is required to perform non-protocol modifications to the JSON object
+    const tx_ = tx as any
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A transaction JSON must contain TransactionType field
+    if (tx_.TransactionType === 'Payment' && tx_.DeliverMax != null) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A Payment transaction JSON must contain Amount field
+      if (tx_.Amount == null) {
+        // If only DeliverMax is provided, use it to populate the Amount field
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access -- ensure that the aliased fields are identical
+        tx_.Amount = tx_.DeliverMax
+      }
+
+      // If Amount is not identical to DeliverMax, throw an error
+      if (
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction JSON could contain these fields
+        tx_.Amount != null &&
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction JSON could contain these fields
+        tx_.Amount !== tx_.DeliverMax
+      ) {
+        throw new ValidationError(
+          'PaymentTransaction: Amount and DeliverMax fields must be identical when both are provided',
+        )
+      }
+
+      // remove the DeliverMax field
+
```

Comment:
Unnecessary comment. `delete tx_.DeliverMax` is already concise.

Reply:
fixd in [e27d23e](https://github.com/XRPLF/xrpl.js/pull/2689/commits/e27d23e200f07a3c315bbe9c0a0ca63a15dd2269)